src/iceberg_functions/iceberg_multi_file_reader.cpp

```diff
@@ -187,23 +187,33 @@ unique_ptr<NodeStatistics> IcebergMultiFileList::GetCardinality(ClientContext &c
 	return make_uniq<NodeStatistics>(cardinality, cardinality);
 }
 
-static void DeserializeBounds(const string &lower_bound, const string &upper_bound,
-                              const IcebergColumnDefinition &column, IcebergPredicateStats &out) {
-	string_t lower_bound_blob(lower_bound.data(), lower_bound.size());
-	string_t upper_bound_blob(upper_bound.data(), upper_bound.size());
-	auto deserialized_lower_bound = IcebergValue::DeserializeValue(lower_bound_blob, column.type);
-	auto deserialized_upper_bound = IcebergValue::DeserializeValue(upper_bound_blob, column.type);
-
-	if (deserialized_lower_bound.HasError()) {
-		throw InvalidConfigurationException("Column %s lower bound deserialization failed: %s", column.name,
-		                                    deserialized_lower_bound.GetError());
-	}
-	if (deserialized_upper_bound.HasError()) {
-		throw InvalidConfigurationException("Column %s upper bound deserialization failed: %s", column.name,
-		                                    deserialized_upper_bound.GetError());
-	}
-	out.lower_bound = deserialized_lower_bound.GetValue();
-	out.upper_bound = deserialized_upper_bound.GetValue();
+static void DeserializeBounds(const Value &lower_bound, const Value &upper_bound, const string &name,
+                              const LogicalType &type, IcebergPredicateStats &out) {
+	if (lower_bound.IsNull()) {
+		out.lower_bound = Value(type);
+	} else {
+		D_ASSERT(lower_bound.type().id() == LogicalTypeId::BLOB);
+		auto lower_bound_blob = lower_bound.GetValueUnsafe<string_t>();
+		auto deserialized_lower_bound = IcebergValue::DeserializeValue(lower_bound_blob, type);
+		if (deserialized_lower_bound.HasError()) {
+			throw InvalidConfigurationException("Column %s lower bound deserialization failed: %s", name,
```

Comment:
Note sure about `InvalidConfigurationException`, I would much prefer an `IOException`

Reply:
@Tmonster we had some run-ins before with the use of IOException being a source of contention

Do you agree with using InvalidConfiguration here or is IO suitable here?

Reply:
Yea, mother duck was complaining that throwing an IOException everywhere made it hard to distinguish between system-level I/O failures in duckdb vs malformed Iceberg configurations. 

We may have corrected too hard by putting InvalidConfiguration in a lot of places. Here we are reading a parquet file, so I think IOException is ok.

[Original Issue](https://github.com/duckdb/duckdb-iceberg/pull/134)

Reply:
We're not, we're deserializing a blob coming from either the metadata, the manifest_file or the data_file